RecoEgamma/EgammaTools/src/LowPtConversion.cc

```diff
@@ -0,0 +1,252 @@
+    throw cms::Exception("InvalidHandle", os.str());
+  }
+
+  // Iterate through conversions and calculate quantities (requirement from Nancy)
+  for (const auto& conv : *conversions) {
+    // Filter
+    if (conv.tracks().size() != 2) {
+      continue;
+    }
+
+    // Quality
+    info.valid = conv.conversionVertex().isValid();                                                         // (=true)
+    info.chi2prob = ChiSquaredProbability(conv.conversionVertex().chi2(), conv.conversionVertex().ndof());  // (<0.005)
+    info.quality_high_purity = conv.quality(reco::Conversion::highPurity);                                  // (=true)
+    info.quality_high_efficiency = conv.quality(reco::Conversion::highEfficiency);                          // (none)
+
+    // Tracks
+    info.ntracks = conv.tracks().size();  // (=2)
+    info.min_trk_pt = -1.;                // (>0.5)
+    for (const auto& trk : conv.tracks()) {
+      if (trk.isNonnull() && trk.isAvailable() && (info.min_trk_pt < 0. || trk->pt() < info.min_trk_pt)) {
+        info.min_trk_pt = trk->pt();
+      }
+    }
+    info.ilead = -1;
+    info.itrail = -1;
+    if (conv.tracks().size() == 2) {
+      edm::RefToBase<reco::Track> trk1 = conv.tracks().front();
+      edm::RefToBase<reco::Track> trk2 = conv.tracks().back();
+      if (trk1.isNonnull() && trk1.isAvailable() && trk2.isNonnull() && trk2.isAvailable()) {
+        if (trk1->pt() > trk2->pt()) {
+          info.ilead = 0;
+          info.itrail = 1;
+        } else {
+          info.ilead = 1;
+          info.itrail = 0;
+        }
+      }
+    }
+
+    // Transverse displacement (with respect to beamspot) and vertex radius
+    math::XYZVectorF p_refitted = conv.refittedPairMomentum();
+    float dx = conv.conversionVertex().x() - beamSpot->x0();
+    float dy = conv.conversionVertex().y() - beamSpot->y0();
+    info.l_xy = (p_refitted.x() * dx + p_refitted.y() * dy) / p_refitted.rho();
+    info.vtx_radius = sqrt(conv.conversionVertex().position().perp2());  // (1.5<r<4.)
+
+    // invariant mass from track pair from conversion
+    info.mass_from_conv = conv.pairInvariantMass();
+
+    // Invariant mass from Pin before fit to common vertex
+    if (conv.tracksPin().size() >= 2 && info.ilead > -1 && info.itrail > -1) {
+      math::XYZVectorF lead_Pin = conv.tracksPin().at(info.ilead);
+      math::XYZVectorF trail_Pin = conv.tracksPin().at(info.itrail);
+      info.mass_from_Pin = mee(lead_Pin.x(), lead_Pin.y(), lead_Pin.z(), trail_Pin.x(), trail_Pin.y(), trail_Pin.z());
+      // Opening angle
+      info.delta_cot_from_Pin = 1. / tan(trail_Pin.theta()) - 1. / tan(lead_Pin.theta());
+    }
+
+    // Invariant mass before fit to common vertex
+    if (conv.tracks().size() >= 2 && info.ilead > -1 && info.itrail > -1) {
+      edm::RefToBase<reco::Track> lead_before_vtx_fit = conv.tracks().at(info.ilead);
+      edm::RefToBase<reco::Track> trail_before_vtx_fit = conv.tracks().at(info.itrail);
```

Comment:
```suggestion
      auto lead_before_vtx_fit = conv.tracks().at(info.ilead)->momentum();
      auto trail_before_vtx_fit = conv.tracks().at(info.itrail)->momentum();
```
will require less pointer following

Reply:
Done.